lib/charms/observability_libs/v0/cert_handler.py

```diff
@@ -111,7 +122,9 @@ def __init__(
         self.cert_subject = charm.unit.name.replace("/", "-") if not cert_subject else cert_subject
 
         # Use fqdn only if no SANs were given, and drop empty/duplicate SANs
-        self.sans_dns = list(set(filter(None, (extra_sans_dns or [socket.getfqdn()]))))
+        sans = list(set(filter(None, (extra_sans_dns or [socket.getfqdn()]))))
```

Comment:
minor code readability nit:

```suggestion
        if extra_sans_dns:
            sans = filter(None, extra_sans_dns)
        else:
            sans = [socket.getfqdn()]
        
        self.sans_ips = sans_ips = []
        self.sans_dns = sans_dns = []
        
        for name in sans:
            if is_ip_address(name):
                sans_ips.append(name)
            else:
                sans_dns.append(name)
```

Reply:
I see what you mean.
I'm on the fence on this one because:
- filtering with a predicate is a well established construct that conveys intention effectively
- 3 no-so-long lines vs. 12 lines